File: app/src/main/java/com/example/josegarcia/todaymeal/model/Ingredient.java
package com.example.josegarcia.todaymeal.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.Objects;
import java.util.Queue;

public class Ingredient implements Parcelable {
    private float quantity;
    private String measure;
    @SerializedName("ingredient")
    private String name;

    public Ingredient(int quantity, String measure, String name) {
        this.quantity = quantity;
        this.measure = measure;
        this.name = name;
    }

    protected Ingredient(Parcel in) {
        quantity = in.readFloat();
        measure = in.readString();
        name = in.readString();
    }

    public static final Creator<Ingredient> CREATOR = new Creator<Ingredient>() {
        @Override
        public Ingredient createFromParcel(Parcel in) {
            return new Ingredient(in);
        }

        @Override
        public Ingredient[] newArray(int size) {
            return new Ingredient[size];
        }
    };

    public float getQuantity() {
        return quantity;
    }

    public String getMeasure() {
        return measure;
    }

    public String getName() {
        return name;
    }

    public void setQuantity(float quantity) {
        this.quantity = quantity;
    }

    public void setMeasure(String measure) {
        this.measure = measure;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getLenght() {
        return Float.toString(quantity).length() + name.length();
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (!(obj instanceof Ingredient)) {
            return false;
        }
        Ingredient ingredient = (Ingredient) obj;
        return quantity == ingredient.getQuantity() &&
                measure.equals(ingredient.getMeasure()) &&
                name.equals(ingredient.getName());

    }

    @Override
    public int hashCode() {
        return Objects.hash(quantity, measure, name);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeFloat(quantity);
        parcel.writeString(measure);
        parcel.writeString(name);
    }
}

File: app/src/main/java/com/example/josegarcia/todaymeal/model/Ingredient.kt
package com.example.josegarcia.todaymeal.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import java.util.*

class Ingredient : Parcelable {
    val quantity: Float
    val measure: String
    @SerializedName("ingredient")
    val name: String

    val length: Int
        get() = quantity.toString().length + name.length

    constructor(quantity: Int, measure: String, name: String) {
        this.quantity = quantity.toFloat()
        this.measure = measure
        this.name = name
    }

    constructor(`in`: Parcel) {
        quantity = `in`.readFloat()
        measure = `in`.readString()
        name = `in`.readString()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Ingredient) {
            return false
        }
        return quantity == other.quantity &&
                measure == other.measure &&
                name == other.name
    }

    override fun hashCode() = Objects.hash(quantity, measure, name)

    override fun describeContents() = 0

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeFloat(quantity)
        parcel.writeString(measure)
        parcel.writeString(name)
    }

    companion object CREATOR : Parcelable.Creator<Ingredient> {
        override fun createFromParcel(parcel: Parcel): Ingredient {
            return Ingredient(parcel)
        }

        override fun newArray(size: Int): Array<Ingredient?> {
            return arrayOfNulls(size)
        }
    }
}
